app/Util.hs

```diff
@@ -42,7 +43,7 @@ decodeHex = fst . B16.decode . E.encodeUtf8
 
 hashPassword :: T.Text -> BS.ByteString -> T.Text
 hashPassword password salt =
-     makeHex . SHA.finalize $ SHA.updates SHA.init [salt, E.encodeUtf8 password]
+     makeHex $ BA.convert . Hash.hashFinalize $ Hash.hashUpdates (Hash.hashInitWith Hash.SHA512) [salt, E.encodeUtf8 password]
```

Comment:
The first function application ($) can be written as function composition (.).
I think while we're at it, we should switch to a KDF like Scrypt, which is implemented in Cryptonite as well.

Reply:
Initial implementation to be found in 6d2931b.